enumerables.rb

```diff
@@ -0,0 +1,122 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    conv = if is_a? Range
+             to_a
+           else
+             self
+           end
+    i = 0
+    while i < conv.length
+      yield conv[i]
+      i += 1
+    end
+    conv
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    conv = if is_a? Range
+             to_a
+           else
+             self
+           end
+    i = 0
+    while i < conv.length
+      yield conv[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    selection = []
+    my_each { |item| selection << item if yield(item) }
+    selection
+  end
+
+  def my_all?(arg = nil)
+    check = true
+    if arg
+      my_each { |i| check = false unless arg === i } # rubocop:disable Style/CaseEquality
+    else
+      my_each { |element| check = false unless yield(element) }
+    end
+    check
+  end
+
+  def my_any?(arg = nil)
+    check = false
+    if arg
+      my_each { |i| check = true if arg === i } # rubocop:disable Style/CaseEquality
+    elsif block_given?
+      my_each { |element| check = true if yield(element) }
+    else
+      my_each { |element| check = true if element }
+    end
+    check
+  end
+
+  def my_none?(arg = nil)
+    check = true
+    if arg
+      my_each { |i| check = false if arg === i } # rubocop:disable Style/CaseEquality
+    elsif block_given?
+      my_each { |element| check = false if yield(element) }
+    else
+      my_each { |element| check = false if element }
+    end
+    check
+  end
```

Comment:
[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:
```ruby
  def my_none?(arg = nil, &block)
    !my_any?(arg, &block)
  end
```
To make this work we have to use block.call(element) instead of yield(element) in my_any?

Reply:
Hi Maria,

Thanks for the review.

I have fixed the problem and would request for a review once again.